keras/applications/regnet.py

```diff
@@ -0,0 +1,1285 @@
+                         strides=stride, use_bias=False)(inputs)
+    skip = layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(skip)
+    conv_3x3 = layers.Conv2D(filters_out, (3, 3), use_bias=False, strides=stride,
+                             groups=groups)
+  else:
+    skip = inputs
+    conv_3x3 = layers.Conv2D(filters_out, (3, 3), use_bias=False,
+                             groups=groups)
+
+  # Build block
+  # conv_1x1_1
+  x = layers.Conv2D(filters_out, (1, 1), use_bias=False)(inputs)
+  x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+  x = relu1x1(x)
+
+  # # conv_3x3
+  x = conv_3x3(x)
+  x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+  x = relu3x3(x)
+
+  # SE 
+  x = layers.GlobalAveragePooling2D()(x)
+  x = layers.Reshape((1, 1, filters_out))(x)
+  x = layers.Conv2D(se_filters, (1, 1), activation="relu")(x)
+  x = layers.Conv2D(filters_out, (1, 1), activation="sigmoid")(x)
+
+
+  # conv_1x1_2
+  x = layers.Conv2D(filters_out, (1, 1), use_bias=False)(x)
+  x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+
+  x = relu(x + skip)
+
+  return x
+
+
+def ZBlock(inputs,
+           filters_in,
+           filters_out,
+           group_width,
+           stride=1,
+           se_ratio=0.25,
+           b=0.25):
+  """Implementation of Z block
+  Reference: [Fast and Accurate Model Scaling](https://arxiv.org/abs/2103.06877)
+  Note that Z block can be completely 
+  
+  Args:
+    inputs: input tensor
+    filters_in: Filters in the input tensor
+    filters_out: Filters in the output tensor
+    group_width: Group width
+    stride: Stride
+    se_ratio: Expansion ration for Squeeze and Excite block
+    b: inverted bottleneck ratio 
+  Return:
+    Output tensor of the block 
+  """
+
+  if filters_in != filters_out and stride == 1:
+    raise ValueError("""Input filters and output filters are not equal for stride
+                         1. Please check inputs and try again.""")
+  
+  groups = filters_out // group_width
+  se_filters = int(filters_out * se_ratio)
+
+  inv_btlneck_filters = int(filters_out / b)
+  if stride != 1:
+    conv_3x3 = layers.Conv2D(inv_btlneck_filters, (3, 3), use_bias=False, strides=stride,
+                             groups=groups)
+  else:
+    conv_3x3 = layers.Conv2D(inv_btlneck_filters, (3, 3), use_bias=False,
+                             groups=groups)
+
+  # Build block
+  # conv_1x1_1
+  x = layers.Conv2D(inv_btlneck_filters, (1, 1), use_bias=False)(inputs)
+  x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+  x = tf.nn.silu(x)
+
+  # # conv_3x3
+  x = conv_3x3(x)
+  x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+  x = tf.nn.silu(x)
+
+  # SE
+  x = layers.GlobalAveragePooling2D(name="Y_GlobalAvgPool")(x)
+  x = layers.Reshape((1, 1, inv_btlneck_filters))(x)
+  x = layers.Conv2D(se_filters, (1, 1), activation=tf.nn.silu)(x)
+  x = layers.Conv2D(inv_btlneck_filters, (1, 1), activation="sigmoid")(x)
```

Comment:
"Squeeze-Excitation block" along with a link to the paper.

Reply:
@sayakpaul

Please feel free to review again. I've made the changes, however I'm not able to re-request a review.